tests/tests_perf/test_perf_utils.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# *****************************************************************************
+# Copyright (c) 2016-2020, Intel Corporation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# - Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import time
+
+import pandas
+
+
+def get_exec_times(f, *args, repeat=5, number=1000000):
+    """
+    Get execution times.
+
+    Parameters
+    ----------
+    f : func
+        function to execute
+    args : tuple
+        parameters of the fucntion
+    repeat : int
+        number of measurements
+    number : int
+        number of the function calls within a single measurement
+
+    Returns
+    -------
+    list
+        list of execution times
+    """
+    exec_times = []
+
+    # Warming up
+    f(*args)
+
+    for _ in range(repeat):
+        start = time.time()
+        for _ in range(number):
+            f(*args)
+        finish = time.time()
```

Comment:
Not sure it is good idea. It gives us average among many function calls. I'm not sure about it. need to think.

Reply:
You are right. So I applied module `timeit` to measure execution times. This module also uses parameters `repeat` and `number`. The parameter `number` doesn't average time among many function calls and just returns full time of all the function calls.

Reply:
Moreover we are able to change the parameters values in the tests. So if it needed we can call the function only once.

Reply:
Current example of the results:
```
Performance testing results:
                                   min       max    median
name   lib   dtype   size
matmul dpnp  float32 262144  1.880e-03 1.915e-03 1.882e-03
                     1048576 8.903e-02 1.838e-01 1.008e-01
                     4194304 4.716e-01 5.019e-01 4.901e-01
             float64 262144  2.743e-03 2.784e-03 2.750e-03
                     1048576 1.338e-01 3.595e-01 3.000e-01
                     4194304 8.860e-01 1.142e+00 9.886e-01
             int32   262144  2.822e-02 4.114e-02 2.903e-02
                     1048576 9.454e-02 9.820e-02 9.774e-02
                     4194304 3.919e-01 3.968e-01 3.937e-01
             int64   262144  2.111e-02 2.403e-02 2.141e-02
                     1048576 9.981e-02 1.012e-01 1.000e-01
                     4194304 4.261e-01 4.317e-01 4.280e-01
       numpy float32 262144  1.760e-03 1.786e-03 1.763e-03
                     1048576 2.955e-02 9.488e-02 6.367e-02
                     4194304 5.124e-01 6.166e-01 5.485e-01
             float64 262144  1.772e-03 1.828e-03 1.779e-03
                     1048576 3.245e-02 7.371e-02 4.669e-02
                     4194304 4.638e-01 6.217e-01 5.696e-01
             int32   262144  1.452e-02 1.457e-02 1.454e-02
                     1048576 6.012e-02 1.048e-01 6.061e-02
                     4194304 3.205e-01 3.313e-01 3.290e-01
             int64   262144  1.490e-02 1.537e-02 1.503e-02
                     1048576 9.386e-02 9.508e-02 9.464e-02
                     4194304 4.207e-01 4.335e-01 4.267e-01
```

Now we don't divide execution time by number=100.